diff --git a/src/btor2aiger.cpp b/src/btor2aiger.cpp
index cef9a00..ea25bda 100644
--- a/src/btor2aiger.cpp
+++ b/src/btor2aiger.cpp
@@ -57,7 +57,7 @@ class Btor2Model
   Btor *btor;
 
   std::vector<BoolectorNode *> inputs;
-  std::unordered_map<int64_t, BoolectorNode *> states;
+  std::vector<std::pair<int64_t, BoolectorNode *>> states;
   std::unordered_map<int64_t, BoolectorNode *> init;
   std::unordered_map<int64_t, BoolectorNode *> next;
   std::vector<BoolectorNode *> bad;
@@ -252,7 +252,7 @@ parse_btor2 (FILE *infile, Btor2Model &model)
         if (l->tag == BTOR2_TAG_input)
           model.inputs.push_back (node);
         else
-          model.states[l->id] = node;
+          model.states.push_back (std::make_pair (l->id, node));
         break;
 
       case BTOR2_TAG_next:
